### DISCUSSION TITLE
Load .mbtiles

### DISCUSSION THREAD

**Discussion by Stanisla** [maplibre/maplibre-gl-js#5177] [General]
Hi all.
I'm designing an app with Cordova for show maps offline.
I'm searching a way for render local .mbtiles inside my app.Searching over the web i've found this: [stackoverflow question](https://stackoverflow.com/questions/720419/how-can-i-find-out-whether-a-server-supports-the-range-header)
In the answer @HarelM shows how he has done.If i understand it correctly,uses `maplibregl.addProtocol`.
Can anyone post a bare example about how implement it?
Thanks in advance.

**Comment by HarelM** [comment]
I have since moved to use PMTiles, it's much easier and doesn't require a sqlite plugin for cordova/capacitor.
See here:
https://github.com/IsraelHikingMap/Site/blob/1715972ea2646e25fc38a34278df9642b91ba88e/IsraelHiking.Web/src/application/services/database.service.ts#L92

**Comment by Stanisla** [comment]
Even better!
Sorry if i asking too much.This was my initial attempt.
With Cordova running as browser ( cordova run browser ) i've not problems.
While i try running as electron or android i get errors about headers:
`Error: Server returned no content-length header or content-length exceeding request. Check that your storage backend supports HTTP Byte Serving`
I've seen your question in pmtiles repository [https://github.com/protomaps/PMTiles/issues/29](https://github.com/protomaps/PMTiles/issues/29) but don't know how to do.
Should i close this thread and open a new one relative to pmtiles?
Thanks again.

**Reply by HarelM** [reply]
The code I've linked is a code that works in my app in production. You'll need the `cordova-plugin-file` to read parts for the file saved in the data directory as requested by pmtiles architecture.
Good luck!

**Comment by Stanisla** [comment]
Thanks HareIM.
Finally i get a working version in vanilla JS.
For anyone interested here is a repo with very basic funcionality (just showing the map)
[https://github.com/Stanisla/CordovaPmtiles/](url)
